src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -495,92 +503,87 @@ static void getOntimeIntervalsThreads(std::string tenant, std::vector<BSONObj>&
-    //
-    // Now, if 'sub.getField("_id")' is not found, if we continue, calling OID() on it, then we get 
-    // an exception and the broker crashes. 
-    //
-    if (idField.eoo() == true)
+    if (strcmp(STR_FIELD(condition, CSUB_CONDITIONS_TYPE).c_str(), ON_TIMEINTERVAL_CONDITION) == 0)
     {
-      LM_E(("Database Error (error retrieving _id field in doc: %s)", sub.toString().c_str()));
-      continue;
-    }
-
-    std::string subId = idField.OID().str();
+      int interval = condition.getIntField(CSUB_CONDITIONS_VALUE);
 
-    std::vector<BSONElement> condV = sub.getField(CSUB_CONDITIONS).Array();
-    for (unsigned int ix = 0; ix < condV.size(); ++ix) {
-      BSONObj condition = condV[ix].embeddedObject();
-      if (strcmp(STR_FIELD(condition, CSUB_CONDITIONS_TYPE).c_str(), ON_TIMEINTERVAL_CONDITION) == 0) {
-        int interval = condition.getIntField(CSUB_CONDITIONS_VALUE);
-        LM_T(LmtNotifier, ("creating ONTIMEINTERVAL for subscription %s with interval %d (tenant %s)", subId.c_str(), interval, tenant.c_str()));
-        processOntimeIntervalCondition(subId, interval, tenant);
-      }
+      LM_T(LmtNotifier, ("creating ONTIMEINTERVAL thread for subscription '%s' with interval %d (tenant '%s')",
+                         subId.c_str(),
+                         interval,
+                         tenant.c_str()));
+      processOntimeIntervalCondition(subId, interval, tenant);
     }
   }
 }
 
 /* ****************************************************************************
 *
-* destroyAllOntimeIntervalThreads -
-*
-* This function is only to be used under harakiri mode, not for real use
+* recoverOntimeIntervalThreads -
 */
-extern void destroyAllOntimeIntervalThreads(std::string tenant)
+void recoverOntimeIntervalThreads(std::string tenant)
 {
-  /* Look for all ONTIMEINTERVAL ids on the given tenant */
-  std::vector<BSONObj> ids;
-  getOntimeIntervalsThreads(tenant, ids);
-
-  /* Iterate destroying thread by thread */
-  for (unsigned int ix = 0; ix < ids.size(); ++ix)
-  {
-    BSONObj      sub     = ids[ix];
-    BSONElement  idField = sub.getField("_id");
-
-    //
-    // BSONElement::eoo returns true if 'not found', i.e. the field "_id" doesn't exist in 'sub'
-    //
-    // Now, if 'sub.getField("_id")' is not found, if we continue, calling OID() on it, then we get 
-    // an exception and the broker crashes. 
-    //
-    if (idField.eoo() == true)
-    {
-      LM_E(("Database Error (error retrieving _id field in doc: %s)", sub.toString().c_str()));
-      continue;
-    }
+  treatOnTimeIntervalSubscriptions(tenant, recoverOnTimeIntervalThread);
+}
 
-    std::string  subId = idField.OID().str();
+/* ****************************************************************************
+*
+* destroyOnTimeIntervalThread - 
+*/
+static void destroyOnTimeIntervalThread(std::string tenant, BSONObj* subP)
+{
+  BSONElement  idField = subP->getField("_id");
+  std::string  subId   = idField.OID().str();
```

Comment:
Well, those checks were there already and I decided to let them stay.
I can put one in here too if you want :-) (probably not necessary but it doesn't hurt)